doc/proxy-clients.rst

```diff
@@ -101,12 +101,24 @@ You can also pass some options to the Varnish client:
   which tags to invalidate when sending invalidation requests to the caching
   proxy. Make sure that your :ref:`Varnish configuration <varnish_tagging>`
   corresponds to the header used here;
+* ``tag_mode`` (ban): Makes it possible to invalidate tags
+  using purge when for instance using ``xkey`` instead of ban. Supports three
+  modes: ``ban``, ``purge``, and ``purge_single`` for purge per tag.
 * ``header_length`` (default: 7500): Control the maximum header length when
   invalidating tags. If there are more tags to invalidate than fit into the
   header, the invalidation request is split into several requests;
 * ``default_ban_headers`` (default: []): Map of headers that are set on each
   ban request, merged with the built-in headers.
 
+Example for configuring xkey::
```

Comment:
can we make this into a subsection "Configuring for xkey"?

if i understand correctly, you also need to change the cache tagging listener to use the header name `key` as that header is hardcoded in xkey. we should explain this here. (and maybe have a very short mention in the doc on the cache header option of the tag listener that references this subsection we create here: "See 'Configuring for xkey' if you want to use the Varnish xkey module")

Reply:
Could you check this again and expand on what you wanted here? (as in, there is already an example here, should I change it so both have own sub header here? )

Reply:
my suggestion was that we group everything xkey related to its own section so we have a place to explain how the whole thing works together. i think we have most or all of the documentation somewhere, but its spread in between other things.